### DISCUSSION TITLE
Setting state/numbers to existing value

### DISCUSSION THREAD

**Discussion by iancg** [nathanmarlor/foxess_modbus#626] [General]
Hi,

I have some automations which are triggered by multiple triggers, and as such may try and say change the inverter mode to backup or feed-in first when the inverter is already in that mode from some other trigger.

Is this harmless or should I always include in my automation

if mode != desired mode then
   set mode to desired mode

?

Same question for setting numbers (e.g. Force Charge Power) - should I wrap the set with a condition to make sure it isn't already that value or is this just a waste of time?

**Comment by FozzieUK** [comment]
That's a good question and I wish I knew the absolute answer - I always do the check before setting a mode or number but I think only a change in value would result in any modbus activity. Hopefully @canton7 can confirm, it'll make my code a lot easier to maintain without the checks.

**Reply by canton7** [reply]
I'd have to check too - it probably comes down to whether HA notifies us about it at all.

But even if we do an unnecessary write, who cares? We're reading hundreds of registers even 10s. Unless you're writing several a second, I wouldn't worry tbh.

**Reply by FozzieUK** [reply]
No worries, thanks 👍 

As this has been niggling me for a while I tested it, I used mode select and number set which covers the majority of our interactions with the integration to see what HA does with these.

Using 'Select: Select', 'Work Mode', 'Feed-in First' - it **always** writes to the 41000 register even if the mode is already set to 'Feed-in First'

Using 'Number: Set', 'Max Charge Current' - it **always** writes to the 41007 register even if the charge current is the same.

As you say, a few writes isn't going to make much difference but at least we now know 😌
